src/bin/dijkstra.rs

```diff
@@ -0,0 +1,203 @@
+// Dijkstra path planning
+// author: Salah Eddine Ghamri (s.ghamri)
+
+use gnuplot::*;
+use std::thread::sleep;
+use std::time::Duration;
+
+use itertools::iproduct;
+use std::collections::BTreeMap;
+
+use ordered_float::NotNan;
+use std::cmp::Reverse;
+use std::collections::BinaryHeap;
+
+use na::DMatrix;
+extern crate nalgebra as na;
+use rust_robotics::grid_map;
+
+fn neighbors(nrows: usize, ncols: usize) -> impl Fn(usize, usize) -> Vec<(usize, usize)> {
+    move |i, j| {
+        let delta = [-1, 0, 1];
+        let neighbors: Vec<_> = iproduct!(delta, delta)
+            .filter_map(|(di, dj)| {
+                let ni = i as i32 + di;
+                let nj = j as i32 + dj;
+                if ni >= 0
+                    && ni < nrows as i32
+                    && nj >= 0
+                    && nj < ncols as i32
+                    && (ni, nj) != (i as i32, j as i32)
+                {
+                    Some((ni as usize, nj as usize))
+                } else {
+                    None
+                }
+            })
+            .collect();
+        neighbors
+    }
+}
+
+fn check_collision(obstacle_map: &grid_map::Map, ni: usize, nj: usize) -> bool {
+    let (height, width) = obstacle_map.shape();
+    let get_neighbors = neighbors(height, width);
+
+    let neighbors = get_neighbors(ni, nj);
```

Comment:
To improve clarity, consider renaming the neighbors function or variable to avoid overlap. For example, the function could be renamed to calculate_neighbors. This change would make the code's intent more clear and reduce potential confusion.